dataprofiler/data_readers/base_data.py

```diff
@@ -93,7 +100,22 @@ def data_format(self, value):
     @property
     def file_encoding(self):
         if not self._file_encoding:
-            return sys.getdefaultencoding()
+            # get system default, but if set to ascii, just update to utf-8
+            file_encoding = "utf-8"
+            try:
+                file_encoding = locale.getpreferredencoding(False)
+            except:
+                file_encoding = sys.getfilesystemencoding()
+            finally:
+                if file_encoding.lower() in ['ANSI_X3.4-1968', 'ascii']:
```

Comment:
can lowercased string be ever in uppercased `'ANSI_X3.4-1968'`

Reply:
fixed